packages/client/src/components/LogEntryModal/LogEntryModal.tsx

```diff
@@ -70,39 +76,65 @@ const ButtonContainer = styled.div`
   }
 `;
 
-export function CreateLogEntryModal({
+export function LogEntryModal({
+  header,
+  initialValues,
   handleClose,
-  handleCreate,
-}: CreateLogEntryProps) {
+  handleSubmit,
+}: LogEntryModalProps) {
   return (
     <Modal>
       <ModalContent>
         <CloseButton type="button" onClick={handleClose}>
           X
         </CloseButton>
-        <Header>Create Log Entry</Header>
+        <Header>{header}</Header>
         <StyledForm
           onSubmit={(event: React.SyntheticEvent) => {
             event.preventDefault();
             const target = event.target as typeof event.target & {
               logDate: { value: string };
               logValue: { value: string };
             };
-            const logEntry = {
+            let logEntry: LogEntryRequest | EditLogEntryRequest = {
               logDate: new Date(target.logDate.value),
               logValue: parseInt(target.logValue.value, 10),
             };
-            handleCreate(logEntry);
+            // Type guard to check if initialValues is an edit request
+            if (
+              initialValues &&
+              'id' in initialValues &&
+              'logId' in initialValues
+            ) {
```

Comment:
I initially had the modal have 2 different modes as the way to know if we were in edit mode or create mode. But then I switched it to accept a header cause that was the only difference between the two modals. Unfortunately down the line I found out I would need to send in id and logId as well and this seemed like a fair enough way to know if the modal is in edit mode. I'm totally fine with changing it back to a flag passed into the modal component so feel free to change how this works!